Tasks/app-store-promote/task.json

```diff
@@ -65,6 +65,34 @@
             "helpMarkDown": "The password of the account used to publish to the Apple App Store.",
             "visibleRule": "authType = UserAndPass"
         },
+                {
+            "name": "isTwoFactorAuth",
+            "type": "boolean",
+            "label": "Use Two-Factor Authentication",
+            "defaultValue": false,
+            "required": false,
+            "helpMarkDown": "Select if the Apple account above has two-factor authentication enabled. (See: https://appleid.apple.com/account/manage -> Security)",
+            "visibleRule": "authType = UserAndPass"
+        },
+        {
+            "name": "appSpecificPassword",
+            "type": "string",
+            "label": "App Specific Password",
+            "defaultValue": "",
+            "isConfidential": true,
+            "required": true,
+            "helpMarkDown": "The supplementary Apple-supplied password used with two-factor authentication enabled (https://appleid.apple.com/account/manage -> Security -> APP-SPECIFIC PASSWORDS)",
+            "visibleRule": "isTwoFactorAuth = true"
+        },
```

Comment:
Not easily.  If we want it to be optionally required for when using service endpoint authentication, then we have to leave it like this.

Thinking about it, I think I could create two fields: 'FASTLANE_SESSION_FOR_ENDPOINT' and 'FASTLANE_SESSION_FOR_USERANDPASS', and control their 'required' separately and get that behavior.